buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1281,20 +1281,20 @@ class BeamModulePlugin implements Plugin<Project> {
         /* include dependencies required by runners */
         //if (runner?.contains('dataflow')) {
         if (runner?.equalsIgnoreCase('dataflow')) {
-          testCompile it.project(path: ":runners:google-cloud-dataflow-java", configuration: 'testCompile')
-          shadow it.project(path: ":runners:google-cloud-dataflow-java:worker:legacy-worker", configuration: 'shadow')
+          testCompile it.project(path: ":runners:google-cloud-dataflow-java", configuration: 'testRuntime')
+          testCompile it.project(path: ":runners:google-cloud-dataflow-java:worker:legacy-worker", configuration: 'testRuntime')
```

Comment:
Since legacy-worker is still shading, you need to use the shadow configuration when referring to it.

```suggestion
          testCompile it.project(path: ":runners:google-cloud-dataflow-java:worker:legacy-worker", configuration: 'shadow')
```